handbook: add forward deployed engineering content (#18336)
PostHog/posthog.com: contents/handbook/forward-deployed-engineering/overview.md

---
title: Forward deployed engineering overview
sidebar: Handbook
showTitle: true
---

> Forward deployed engineering (FDE) is PostHog's team of engineers who work as an extension of the customer's engineering team to get them to value fast. Depending on the engagement, that might mean hands-on work in their codebase (migration, instrumentation, dashboards, experiments) or it might mean getting their foundations right and enabling their team without ever touching code.

Reviewer: ```suggestion
Forward deployed engineering (FDE) is PostHog's team of engineers embedded with customers to get their implementation right. Depending on the engagement, that might mean hands-on work in their codebase (migration, instrumentation, dashboards, experiments) or it might mean getting their foundations right and enabling their team without ever touching code.

What makes this different from [professional services](/handbook/growth/sales/professional-services) is that every engagement compounds. The patterns we see across customers become reusable artifacts, skills, and product improvements that benefit the next customer and the broader PostHog user base. We strive to make each engagement faster and sharper than the last.

FDE isn't simply doing technical execution. AI made outputting code easy, our work is to be providers of good judgement, prioritizing for our team and the customer, helping them find the root of their problem, then planning and executing the best solution that fits their culture.
```

> Every engagement teaches us something we feed back into the product and our tooling, so the next customer starts further ahead than the last. We sit between the customer, [sales and CS](/handbook/forward-deployed-engineering/working-with-sales-and-cs), and [product engineering](/handbook/forward-deployed-engineering/working-with-product-engineering). See [how we work](/handbook/forward-deployed-engineering/how-we-work) for how we operate day to day.

Reviewer: ```suggestion
We sit between the customer, [sales and CS](https://github.com/handbook/forward-deployed-engineering/working-with-sales-and-cs), and [product engineering](https://github.com/handbook/forward-deployed-engineering/working-with-product-engineering). See [how we work](https://github.com/handbook/forward-deployed-engineering/how-we-work) for how we operate day to day.
```

## What an FDE is

> 

Reviewer: ```suggestion

We don't maintain a fixed list of services. If solving a customer's problem produces something reusable, we're interested in seeing if we can help. In practice, that often looks like:
```

- Help customers design and implement PostHog for their specific stack, scale, and use cases
- Unblock technical adoption: instrumentation, data modeling, migrations, and integrations
- Build reference implementations, examples, and internal tooling that make future engagements faster
- Feed real customer problems back into the product via our [product engineering teams](/handbook/forward-deployed-engineering/working-with-product-engineering)

> The throughline: **customer value that compounds.** Anything we solve for one customer should, where possible, become a reusable artifact (a template, a reference, or a product improvement) so the same problem is cheaper to solve next time.

Reviewer: ```suggestion

```

## What an FDE isn't

> - Not support. [Support](/handbook/support/support-team) is reactive and one-off; it answers and moves on. FDE work is scoped, higher-touch, and tied to a specific customer outcome, and it leaves a durable artifact behind.

Reviewer: ```suggestion
- Support. We have an amazing [Support](/handbook/support/support-team) team who provide deep technical insight and fixes issues day-to-day, but they have to move fast from customer to customer. FDE work is higher-touch, tied to a specific customer outcome, and designed to leave something reusable behind.
```

> - Not sales engineering. Pre-sales is time-boxed to help close a deal. FDE engagements begin where that ends.

Reviewer: ```suggestion
- Sales, success, or account management. Your TAM or CSM owns the commercial relationship, we're embedded on the technical side for scoped engagements.
```

> - Not traditional professional services. ProServ is scoped, billed work for one customer that doesn't generalize. FDEs deliver that work, but the role is defined by the part that *does* generalize.

Reviewer: ```suggestion
- Product engineering. Our focus is work that compounds across customers and engagements, not owning product direction, though the insights we surface help our team prioritize.
```

The [engagement types](/handbook/forward-deployed-engineering/how-we-work#engagement-types) section makes these distinctions concrete.

## When FDE gets involved

> FDE is a finite resource, so we're deliberate about where we spend time. Typical triggers:

Reviewer 2: ```suggestion
The FDE team is small so we're deliberate about where we spend time. Typical triggers:
```

Don't say resource - corpo speak
